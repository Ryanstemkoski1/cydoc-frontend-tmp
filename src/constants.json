{
  "medicalhistory": {
    "conditions": ["Type II Diabetes", "Myocardial Infarction", "Hypertension",  "Hypercholesteremia", "Depression", "HIV"]
  },

  "socialhistory": {
    "substanceUseContentHeader": "Substance Use",
    "secondaryFields": ["living situation", "employment", "diet", "exercise"],
    "substanceAbuseFields" : {
      "tobacco": {
        "condition": "tobacco",
        "firstField": "Packs/Day",
        "secondField": "Number of Years"
      },
      "alcohol": {
        "condition": "alcohol",
        "firstField": "Drinks/Week",
        "secondField": "What kind of drinks?"
      },
      "substanceAbuse": {
        "condition": "substance abuse",
        "firstField": "Drug Name",
        "secondField": "Mode of Delivery"
      }
    }
  }

}