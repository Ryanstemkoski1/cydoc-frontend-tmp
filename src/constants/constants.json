{
  "TAB_NAMES": [
    "HPI",
    "Patient History",
    "Review of Systems",
    "Physical Exam",
    "Plan",
    "Generate Note"
  ],
  "CONDITIONS": [
    "Type II Diabetes",
    "Myocardial Infarction",
    "Hypertension",
    "Hypercholesteremia",
    "Depression",
    "HIV"
  ],
  "MEDICAL_HISTORY": {
    "STATE": {
      "0": {
        "Condition": "Type II Diabetes",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "1": {
        "Condition": "Myocardial Infarction",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "2": {
        "Condition": "Hypertension",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "3": {
        "Condition": "Hypercholesteremia",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "4": {
        "Condition": "Depression",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "5": {
        "Condition": "HIV",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      }
    }
  },
  "FAMILY_HISTORY": {
    "STATE": {
      "0": {
        "Condition": "Type II Diabetes",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "1": {
        "Condition": "Myocardial Infarction",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "2": {
        "Condition": "Hypertension",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "3": {
        "Condition": "Hypercholesteremia",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "4": {
        "Condition": "Depression",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "5": {
        "Condition": "HIV",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      }
    }
  },
  "SOCIAL_HISTORY": {
    "SUBSTANCE_USE_CONTENT_HEADER": "Substance Use",
    "SECONDARY_FIELDS": [
      "Living Situation",
      "Employment",
      "Diet",
      "Exercise"
    ],
    "SUBSTANCE_USE_FIELDS": {
      "Tobacco": {
        "condition": "Tobacco",
        "firstField": "Packs/Day",
        "secondField": "Number of Years",
        "thirdField": "Products Used",
        "quitYear": "Quit Year"
      },
      "Alcohol": {
        "condition": "Alcohol",
        "firstField": "Drink Type",
        "secondField": "Drink Size",
        "thirdField": "# Per Week",
        "quitYear": "Quit Year"
      },
      "Substance Abuse": {
        "condition": "Substance Abuse",
        "firstField": "Drug Name",
        "secondField": "Mode of Delivery",
        "thirdField": "# Per Week",
        "quitYear": "Quit Year"
      }
    },
    "STATE": {
      "Alcohol": {
        "Yes": false,
        "In the Past": false,
        "Never Used": false,
        "Drinks Type": "",
        "Drink Size": "",
        "# Per Week": "",
        "Quit Year": "",
        "Comments": ""
      },
      "Tobacco": {
        "Yes": false,
        "In the Past": false,
        "Never Used": false,
        "Packs/Day": "",
        "Number of Years": "",
        "Products Used": "",
        "Quit Year": "",
        "Comments": ""
      },
      "Substance Abuse": {
        "Yes": false,
        "In the Past": false,
        "Never Used": false,
        "Drug Name": "",
        "Mode of Delivery": "",
        "# Per Week": "",
        "Quit Year": "",
        "Comments": ""
      },
      "Living Situation": "",
      "Employment": "",
      "Diet": "",
      "Exercise": ""
    }
  },
  "ALLERGIES": {
    "fields": [
      "Inciting Agent",
      "Reaction",
      "Comments"
    ],
    "STATE": {
      "Inciting Agent": "",
      "Reaction": "",
      "Comments": ""
    }
  }
}