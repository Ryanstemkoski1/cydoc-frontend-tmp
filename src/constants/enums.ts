export enum YesNoResponse {
    Yes = 'YES',
    No = 'NO',
    None = '',
}

export enum YesNoMaybeResponse {
    Yes = 'YES',
    No = 'NO',
    Maybe = 'MAYBE',
    None = '',
}

export enum SubstanceUsageResponse {
    Yes = 'YES',
    InThePast = 'IN_THE_PAST',
    NeverUsed = 'NEVER_USED',
    None = '',
}

export interface LRButtonState {
    left: boolean;
    center: boolean;
    right: boolean;
}

export enum LeftRight {
    Left = 'LEFT',
    Right = 'RIGHT',
}
export enum YesNoUncertainResponse {
    Yes = 'YES',
    No = 'NO',
    Uncertain = 'UNCERTAIN',
    None = '',
}

export enum WhenResponse {
    Today = 'TODAY',
    ThisWeek = 'THIS_WEEK',
    ThisMonth = 'THIS_MONTH',
    ThisYear = 'THIS_YEAR',
    None = '',
}

export interface DropdownOption {
    key: string | number;
    value: string;
    text: string;
}
