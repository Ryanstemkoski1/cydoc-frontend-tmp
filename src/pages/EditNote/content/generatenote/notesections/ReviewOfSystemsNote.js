import React from 'react';

class ReviewOfSystemsNote extends React.Component {
    render() {
        const review = this.props.reviewOfSystems;

        let components = [];
        for (var key in review) {
            let positives = [];
            let negatives = [];
            for (var question in review[key]) {
                if (review[key][question] === 'y') {
                    positives.push(question.toLowerCase());
                } else if (review[key][question] === 'n') {
                    negatives.push(question.toLowerCase());
                }
            }
            components[key] = {
                positives: positives,
                negatives: negatives
            }
        }
        return (
            <ul>
                {Object.keys(components).map(key => (
                    components[key].positives.length > 0 || components[key].negatives.length > 0 ?
                        <li>
                            <b>{key}: </b>
                            {components[key].positives.length > 0 ? `Positive for ${components[key].positives.join(', ')}. `: null}
                            {components[key].negatives.length > 0 ? `Negative for ${components[key].negatives.join(', ')}. ` : null}
                        </li>
                    : null
                ))}
            </ul>
        )
    }
}

export default ReviewOfSystemsNote;