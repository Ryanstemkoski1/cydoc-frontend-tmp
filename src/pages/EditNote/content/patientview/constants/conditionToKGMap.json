{
    "ADHD": "ADHD",
    "Depression": "DEPRESSION",
    "Genital Symptoms": "SEXUALINFECTION",
    "High Blood Pressure": "HYPERTENSION",
    "High Cholesterol": "HYPERLIPIDEMIA",
    "High Triglycerides": "HYPERLIPIDEMIA",
    "Urinary Symptoms": "URINARY_SYMPTOMS"
}
