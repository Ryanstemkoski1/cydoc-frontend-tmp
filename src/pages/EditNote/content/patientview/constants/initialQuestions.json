{
    "order": {
        "1": "1",
        "2": "2",
        "3": "3",
        "4": "4",
        "5": "5",
        "6": "6",
        "7": "7"
    },
    "graph": {
        "1": ["2", "3", "4", "8", "9"],
        "2": [],
        "3": [],
        "4": [],
        "5": ["6", "7"],
        "6": [],
        "7": []
    },
    "nodes" : {
            "1": {
                "uid": "1",
                "text": "",
                "responseType": "",
                "category": "",
                "doctorView": ""
            },
            "2": {
                "uid": "2",
                "text": "Are you here for an annual physical exam?",
                "responseType": "YES-NO",
                "category": "ANNUALVISIT",
                "doctorView": "Annual Physical Exam"
            },
            "3": {
                "uid": "3",
                "text": "Are you here to follow up any previously diagnosed medical conditions?",
                "responseType": "YES-NO",
                "category": "",
                "doctorView": ""
            },
            "4": {
                "uid": "4",
                "text": "Do you have any new health concerns to discuss with the clinician today?",
                "responseType": "YES-NO",
                "category": "",
                "doctorView": ""
            },
            "5": {
                "uid": "5",
                "text": "",
                "responseType": "",
                "category": "",
                "doctorView": ""
            },
            "6": {
                "uid": "6",
                "text": "Please select the most important conditions or symptoms that you would like to discuss today, up to a maximum of 3. CLICK[ADHD, Depression, Diabetes, Genital Symptoms/STI, Hypertension, Hyperlipidemia, Urinary Symptoms/UTI]",
                "responseType": "SELECTONE",
                "category": "",
                "doctorView": ""
            },
            "7": {
                "uid": "7",
                "text": "If youâ€™d like to discuss an additional condition or symptom not listed above, please search for it here:",
                "responseType": "SEARCH",
                "category": "",
                "doctorView": ""
            },
            "8": {
                "uid": "8",
                "text": "Please confirm the date of your appointment.",
                "responseType": "TIME3DAYS",
                "category": "",
                "doctorView": ""
            },
            "9": {
                "uid": "9",
                "text": "What is the last name of the clinician you are seeing today? (optional)",
                "responseType": "SHORT-TEXT",
                "category": "",
                "doctorView": ""
            }
        }
}
