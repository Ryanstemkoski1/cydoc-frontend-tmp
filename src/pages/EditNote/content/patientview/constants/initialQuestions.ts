import { ResponseTypes } from 'constants/hpiEnums';
import { InitialQuestionsState } from 'redux/reducers/userViewReducer';

const initialQuestions: InitialQuestionsState = {
    order: {
        '1': '1',
        '2': '2',
        '3': '3',
        '4': '4',
        '5': '5',
        '6': '6',
        '7': '7',
    },
    graph: {
        '1': ['8', '9'],
        '2': [],
        '3': [],
        '4': [],
        '5': ['6', '7'],
        '6': [],
        '7': [],
    },
    nodes: {
        '1': {
            uid: '1',
            text: '',
            responseType: ResponseTypes.NULL,
            category: '',
            doctorView: '',
        },
        '2': {
            uid: '2',
            text: 'Are you here for an annual physical exam?',
            responseType: ResponseTypes.YES_NO,
            category: 'ANNUALVISIT',
            doctorView: 'Annual Physical Exam',
        },
        '3': {
            uid: '3',
            text: 'Are you here to follow up any previously diagnosed medical conditions?',
            responseType: ResponseTypes.YES_NO,
            category: '',
            doctorView: '',
        },
        '4': {
            uid: '4',
            text: 'Do you have any new health concerns to discuss with the clinician today?',
            responseType: ResponseTypes.YES_NO,
            category: '',
            doctorView: '',
        },
        '5': {
            uid: '5',
            text: '',
            responseType: ResponseTypes.NULL,
            category: '',
            doctorView: '',
        },
        '6': {
            uid: '6',
            text: 'Please select the most important conditions or symptoms that you would like to discuss today, up to a maximum of 3.',
            responseType: ResponseTypes.SELECTONE,
            category: '',
            doctorView: '',
        },
        '7': {
            uid: '7',
            text: 'List any other conditions or symptoms you would like to discuss with the clinician today. Please be accurate as this information will be used to organize your visit.',
            responseType: ResponseTypes.LIST_TEXT,
            category: '',
            doctorView: '',
        },
        '8': {
            uid: '8',
            text: 'Please confirm the date of your appointment.',
            responseType: ResponseTypes.TIME3DAYS,
            category: '',
            doctorView: '',
        },
        '9': {
            uid: '9',
            text: 'What is the last name of the clinician you are seeing today? (optional)',
            responseType: ResponseTypes.SHORT_TEXT,
            category: '',
            doctorView: '',
        },
    },
};

export default initialQuestions;
