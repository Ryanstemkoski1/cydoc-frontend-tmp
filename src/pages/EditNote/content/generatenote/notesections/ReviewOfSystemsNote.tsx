import React, { Component } from 'react';
import { ReviewOfSystemsState } from 'redux/reducers/reviewOfSystemsReducer';
import { YesNoResponse } from 'constants/enums';

interface ROSProps {
    ROSState: ReviewOfSystemsState;
}

export class ReviewOfSystemsNote extends Component<ROSProps> {
    render() {
        const { ROSState } = this.props;
        const review = ROSState;

        const components: {
            [category: string]: {
                [key: string]: string[];
            };
        } = {};
        for (const category in review) {
            const positives: string[] = [];
            const negatives: string[] = [];
            for (const option in review[category]) {
                if (review[category][option] === YesNoResponse.Yes) {
                    positives.push(option.toLowerCase());
                } else if (review[category][option] === YesNoResponse.No) {
                    negatives.push(option.toLowerCase());
                }
            }
            components[category] = {
                positives: positives,
                negatives: negatives,
            };
        }

        let isEmpty = true;
        for (const category in components) {
            if (
                !(
                    components[category].positives.length === 0 &&
                    components[category].negatives.length === 0
                )
            ) {
                isEmpty = false;
            }
        }

        if (isEmpty) {
            return <div>No review of systems reported.</div>;
        }

        return (
            <ul>
                {Object.keys(components).map((key) =>
                    components[key].positives.length > 0 ||
                    components[key].negatives.length > 0 ? (
                        <li>
                            <b>{key}: </b>
                            {components[key].positives.length > 0
                                ? `Positive for ${components[
                                      key
                                  ].positives.join(', ')}. `
                                : null}
                            {components[key].negatives.length > 0
                                ? `Negative for ${components[
                                      key
                                  ].negatives.join(', ')}. `
                                : null}
                        </li>
                    ) : null
                )}
            </ul>
        );
    }
}

export default ReviewOfSystemsNote;
