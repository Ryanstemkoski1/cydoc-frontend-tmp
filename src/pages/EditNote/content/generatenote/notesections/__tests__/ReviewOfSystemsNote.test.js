import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { ReviewOfSystemsNote } from '../ReviewOfSystemsNote';
import { YesNoResponse } from 'constants/enums';

Enzyme.configure({ adapter: new EnzymeAdapter() });

const emptyROS = {
    General: {
        'Weight changes': YesNoResponse.None,
        Fatigue: YesNoResponse.None,
        Weakness: YesNoResponse.None,
        Fevers: YesNoResponse.None,
        Chills: YesNoResponse.None,
        'Night sweats': YesNoResponse.None,
    },
    Eyes: {
        Glasses: YesNoResponse.None,
        Contacts: YesNoResponse.None,
        Blurriness: YesNoResponse.None,
    },
    Ears: {
        'Changes in hearing': YesNoResponse.None,
        'Hearing loss': YesNoResponse.None,
        Tinnitus: YesNoResponse.None,
        Earache: YesNoResponse.None,
        Discharge: YesNoResponse.None,
    },
};

const mountWithProps = (ros = emptyROS) => {
    return mount(<ReviewOfSystemsNote ROSState={ros} />);
};

describe('ReviewOfSystems Note', () => {
    it('renders without crashing', () => {
        const wrapper = mountWithProps();
        expect(wrapper).toBeTruthy();
    });
    it('matches snapshot', () => {
        const wrapper = mountWithProps();
        expect(wrapper.html()).toMatchSnapshot();
    });
    it('renders correctly when completely empty', () => {
        const wrapper = mountWithProps();
        expect(wrapper.text()).toContain('No review of systems reported');
    });
    it('renders only non-empty options', () => {
        const wrapper = mountWithProps({
            General: {
                'Weight changes': YesNoResponse.Yes,
                Fatigue: YesNoResponse.Yes,
                Weakness: YesNoResponse.No,
                Fevers: YesNoResponse.No,
                Chills: YesNoResponse.None,
                'Night sweats': YesNoResponse.None,
            },
            Eyes: {
                // Shouldn't appear because all Nones
                Glasses: YesNoResponse.None,
                Contacts: YesNoResponse.None,
                Blurriness: YesNoResponse.None,
            },
            Ears: {
                'Changes in hearing': YesNoResponse.Yes,
                'Hearing loss': YesNoResponse.None,
                Tinnitus: YesNoResponse.None,
                Earache: YesNoResponse.None,
                Discharge: YesNoResponse.None,
            },
        });

        const expected = [
            'General: Positive for weight changes, fatigue. Negative for weakness, fevers. ',
            'Ears: Positive for changes in hearing. ',
        ];

        const items = wrapper.find('li');
        expect(items).toHaveLength(2);
        items.forEach((node, i) => expect(node.text()).toEqual(expected[i]));
    });
});
