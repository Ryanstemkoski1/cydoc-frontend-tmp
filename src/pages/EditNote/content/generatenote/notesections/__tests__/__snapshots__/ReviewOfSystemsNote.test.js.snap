// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewOfSystems Note matches snapshot 1`] = `"<div>No review of systems reported.</div>"`;
